### ISSUE TITLE
dns lookup upgraded in 1.6 incorrectly attempts to resolve address in case of an error from the lookup callback

### ISSUE THREAD

**Issue by pveller** [axios/axios#6174]
### Describe the bug

TL;DR: if you supply custom DNS `lookup` in 1.6+ and it fails with `ENOTFOUND`, `axios` throws `TypeError: address must be a string` instead

I supply custom DNS `lookup` to precisely control the timeout. I am using https://github.com/szmarczak/cacheable-lookup. It works correctly in axios before 1.6 but fails after because of this change:

![image](https://github.com/axios/axios/assets/498673/50a0e6ba-7fe0-4b03-b5c9-6b05d7b4cfa1)

When the callback returns the `err` as the first argument, the other two can be (and are) `undefined`. Axios 1.6+ attempts to resolve `family` and `address` regardless of `err` and trips over `undefined`.

Take a look:

![image](https://github.com/axios/axios/assets/498673/5ee84a2e-361b-4bd0-a589-85d81fe9208e)

### To Reproduce

Perform axios request against a domain that does not exists with a custom `lookup`. DNS `lookup` fails with `ENOTFOUND` but `axios` then fails with `TypeError: address must be a string`

### Code snippet

```js
// simplified test:


import CacheableLookup from 'cacheable-lookup';
import axios from 'axios';

const dns = new CacheableLookup({ resolver: new Resolver({ timeout: 200 }), maxTtl: 60 * 60, lookup: false });
const controller = new AbortController();

await axios.post('https://no-such-domain-987654.com', {}, {
  signal: controller.signal,
  lookup: dns.lookup.bind(dns),
});
```

### Expected behavior

`axios` should fail with the DNS `lookup` error, not the `TypeError`

### Axios Version

1.6+
